### PR TITLE
refs #2348: Remember expenses scroll list position

### LINKED ISSUE #2348: Remember Scroll Position In List After Navigating Inside An Expense
While scrolling through a long expense list, if we open a particular expense to know the time of transaction then the whole list starts to scroll from the top when we navigate back. Remembering the scroll position in the list will save the user from having to scroll from the top again.

### PR DESCRIPTION
**Pull request by MaratKhakim**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the "Contributing Rules".
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

[Screen_recording_20231010_210433.webm](https://github.com/Ivy-Apps/ivy-wallet/assets/20263744/801d48d0-7b9b-44aa-b140-6641f5aa6b49)

Describe with a few bullets **what's new:**
- Added a composable function to save a list scroll position

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- Should not break anything

**In what cases your code won't work?**

- N/A

## Does this PR closes any GitHub Issues?

- Closes #2348 

> Replace `ISSUE_NUMBER` with your issue number (for example Closes #1234). If you've done that correctly, you'll see the issue title linked when previewing your PR description.

## Troubleshooting CI failures

If you see any of the PR checks failing (❌) go to [Actions](https://github.com/Ivy-Apps/ivy-wallet/actions) and find it there. Or simply click "Details" next to the failed check and explore the logs to see why it has failed.

### Detekt
[Detekt](https://detekt.dev/) is a static code analyzer for Kotlin that we use to enforce code readibility and good practices.

**To run Detekt locally:**
```
./gradlew detekt
```

If the Detekt errors are caused by a legacy code, you can suppress them using a basline.

**Detekt baseline** (not recommended)
```
./scripts/detektBaseline.sh
```

### Lint

We use the [standard Android Lint](https://developer.android.com/studio/write/lint) plus [Slack's compose-lints](https://slackhq.github.io/compose-lints/) as an addition to enforce proper Compose usage.

**To run Lint locally:**
```
./scripts/lint.sh
```

If the Lint errors are caused by a legacy code, you can suppress them using a basline.

**Lint baseline** (not recommended)
```
./scripts/lintBaseline.sh
```

### Unit tests

If this job is failing this means that your changes break an existing unit test. You must identify the failing tests and fix your code.

**To run the Unit tests locally:**
```
./gradlew testDebugUnitTest
```

### CODE REVIEW

File: .DS_Store
Comment (ILIYANGERMANOV) [inline .DS_Store:1]: Cam you remove this file and add it to `.gitignore`

File: screen-home/src/main/java/com/ivy/home/HomeTab.kt [hunk 1]
```diff
@@ -112,7 +112,8 @@ private fun BoxWithConstraintsScope.UI(
                 }
             )
     ) {
-        val listState = rememberLazyListState(
+        val listState = rememberScrollPositionListState(
+            key = "home_lazy_column",
```
Comment (ILIYANGERMANOV) [inline screen-home/src/main/java/com/ivy/home/HomeTab.kt:116]: `rememberScrollPositionListState` must be applied to `:screen-transactions` too

**Review by ILIYANGERMANOV — APPROVED** [review]
Excellent work 👏 I like the fix 💯

File: .DS_Store
Comment (MaratKhakim) [reply .DS_Store:1]: Done

File: screen-home/src/main/java/com/ivy/home/HomeTab.kt (hunk 1 above)
Comment (MaratKhakim) [reply screen-home/src/main/java/com/ivy/home/HomeTab.kt:116]: Done

**Comment by ILIYANGERMANOV** [comment]
Nicely done ✅ Merging after the CI pass

**Pull request merged**